src/std/text/csv.ss

```diff
@@ -0,0 +1,331 @@
+      (add (current-string))
+      (do-fields))
+     ((or (accept-eol port) (accept-eof port))
+      (add (current-string))
+      (done))
+     ((accept-quote port)
+      (cond
+       ((and (csv-unquoted-quotequote?) (accept-quote port))
+        (add-char (csv-quote)) (do-field-unquoted))
+       ((csv-loose-quote?)
+        (do-field-quoted))
+       (else
+        (raise-io-error 'read-csv-line "unexpected quote in middle of field"))))
+     (else
+      (add-char (read-char port))
+      (do-field-unquoted))))
+  (def (end-of-field)
+    (when (csv-skip-whitespace?)
+      (accept-spaces port))
+    (cond
+     ((or (accept-eol port) (accept-eof port))
+      (done))
+     ((accept-separator port)
+      (do-fields))
+     (else
+      (raise-io-error 'read-csv-line "end of field expected"))))
+  (def (add x)
+    (set! fields (cons x fields)))
+  (def (add-char c)
+    (unless (or (csv-allow-binary?) (char-ascii-text? c))
+      (raise-io-error 'read-csv-line "binary data not allowed" c))
+    (write-char c ss))
+  (def (current-string)
+    (get-output-string ss))
+  (def (done)
+    (reverse! fields))
+  (parameterize ((csv-eol-matrix (csv-eol-matrix<-line-endings (csv-line-endings))))
+    (do-fields)))
+
+;; Read lines from PORT in CSV format, using the current syntax parameters.
+;; Return a list of list of strings, one entry for each line,
+;; that contains one entry for each field.
+;; Entries are read as strings;
+;; it is up to you to interpret the strings as whatever you want.
+(def (read-csv-lines port)
+  (validate-csv-parameters)
+  (call-with-list-builder (lambda (c _) (until (accept-eof port) (c (read-csv-line port))))))
+
+;; Open the file designated by the path, using the provided settings if any,
+;; and call read-csv-lines on it.
+(def (read-csv-file path . settings)
+  (call-with-input-file (cons* path: path settings) read-csv-lines))
+
+(def (char-needs-quoting? x)
+  (or (eqv? x (csv-quote))
+      (eqv? x (csv-separator))
+      (eqv? x #\newline)
+      (eqv? x #\return)
+      (not (char-ascii-text? x))))
+
+(def (string-needs-quoting? x)
+  (and (not (string-null? x))
+       (or (char-csv-space? (string-ref x 0))
+           (char-csv-space? (string-ref x (1- (string-length x))))
+           (string-any char-needs-quoting? x))
+       #t))
+
+;; Given a list of LINES, each of them a list of fields, and a PORT,
+;; format those lines as CSV according to the current syntax parameters.
+(def (write-csv-lines lines port)
+  (for-each (cut write-csv-line <> port) lines))
+
+;; Format one line of FIELDS to PORT in CSV format,
+;; using the current syntax parameters.
+(def (write-csv-line fields port)
+  (let loop ((fields fields))
+    (match fields
+      ([first . rest]
+       (write-csv-field first port)
+       (unless (null? rest)
+         (write-char (csv-separator) port))
+       (loop rest))
+      ([] (display (csv-eol) port)))))
+
+(def (write-csv-field field port)
+  (match field
```

Comment:
this will result in a rather non-descript error from match if the field is not a number or symbol/string.
Perhaps we want an `else` clause that errors more informatively?